cmd/main.go

```diff
@@ -225,11 +229,32 @@ func main() {
 			Service:        containers,
 		}
 
-		provider = &cloud.GKE{
+		compute, err := computev1.NewService(context.Background())
+		if err != nil {
+			setupLog.Error(err, "unable to create compute client")
+			os.Exit(1)
+		}
+		reservationsProvider := &cloud.ReservationProvider{
+			Service:   compute,
+			ProjectID: cfg.GCPProjectID,
+			Zone:      cfg.GCPZone,
+		}
+
+		gkeProvider := &cloud.GKE{
 			NodePools:      nodePoolsService,
+			Reservations:   reservationsProvider,
 			ClusterContext: clusterCtx,
 			Recorder:       mgr.GetEventRecorderFor("tpu-provisioner"),
 		}
+		provider = gkeProvider
+
+		if cfg.StaticProvisionReservations != "" {
+			reservations := strings.Split(cfg.StaticProvisionReservations, ",")
+			if err := gkeProvider.StaticallyProvisionNodePools(ctx, reservations); err != nil {
```

Comment:
I don't think we want to incur the startup latency involved in listing and checking all node pools on every provisioner restart. An inactive provisioner will contribute poorly to TPU utilization.

You really want to do this async... I would lean towards the reconciliation/controller pattern: This reconciler would watch a singular ConfigMap or CRD that has reservations defined and ensure they exist from there. It would fire events on that object to report success/failure. This pattern gets you: async by default, built-in reconciliation on some period (i think it defaults to 24hrs), built in error retrying with expo-backoff, and ability to add reservations without restarting provisioner.